src/App/index.js

```diff
@@ -16,23 +25,116 @@ class App extends Component {
     this.setState(({ mapVisible: prev }) => ({ mapVisible: !prev }));
   };
 
+  locationSuccess = pos => {
+    this.setState({
+      latitude: pos.coords.latitude,
+      longitude: pos.coords.longitude,
+      gotLocation: true,
+    });
+  };
+
+  locationError = err => {
+    console.warn(`ERROR(${err.code}): ${err.message}`);
+    this.setState({
+      locationError: true,
+    });
+    setTimeout(() => {
+      this.setState({
+        locationError: false,
+      });
+      return;
+    }, 5000);
+  };
+
+  getLocation = () => {
```

Comment:
Maybe for a later refactor, but I wonder if we can just use the navigator location logic once? We are doing the same thing in here and in the ReportMap.

Reply:
https://github.com/tmikeschu/mogo-reporter/issues/26